WebCryptoAPI/wrapKey_unwrapKey/wrapKey_unwrapKey.https.any.js

```diff
@@ -6,111 +6,113 @@
+            generateParameters: {name: "AES-GCM", length: 128},
+            wrapParameters: {name: "AES-GCM", iv: new Uint8Array(16), additionalData: new Uint8Array(16), tagLength: 128}
+        },
+        {
+            name: "AES-KW",
+            generateParameters: {name: "AES-KW", length: 128},
+            wrapParameters: {name: "AES-KW"}
+        }
+    ];
+
+    var keysToWrapParameters = [
+        {algorithm: {name: "RSASSA-PKCS1-v1_5", modulusLength: 1024, publicExponent: new Uint8Array([1,0,1]), hash: "SHA-256"}, privateUsages: ["sign"], publicUsages: ["verify"]},
+        {algorithm: {name: "RSA-PSS", modulusLength: 1024, publicExponent: new Uint8Array([1,0,1]), hash: "SHA-256"}, privateUsages: ["sign"], publicUsages: ["verify"]},
+        {algorithm: {name: "RSA-OAEP", modulusLength: 1024, publicExponent: new Uint8Array([1,0,1]), hash: "SHA-256"}, privateUsages: ["decrypt"], publicUsages: ["encrypt"]},
+        {algorithm: {name: "ECDSA", namedCurve: "P-256"}, privateUsages: ["sign"], publicUsages: ["verify"]},
+        {algorithm: {name: "ECDH", namedCurve: "P-256"}, privateUsages: ["deriveBits"], publicUsages: []},
+        {algorithm: {name: "Ed25519" }, privateUsages: ["sign"], publicUsages: ["verify"]},
+        {algorithm: {name: "Ed448" }, privateUsages: ["sign"], publicUsages: ["verify"]},
+        {algorithm: {name: "X25519" }, privateUsages: ["deriveBits"], publicUsages: []},
+        {algorithm: {name: "X448" }, privateUsages: ["deriveBits"], publicUsages: []},
+        {algorithm: {name: "AES-CTR", length: 128}, usages: ["encrypt", "decrypt"]},
+        {algorithm: {name: "AES-CBC", length: 128}, usages: ["encrypt", "decrypt"]},
+        {algorithm: {name: "AES-GCM", length: 128}, usages: ["encrypt", "decrypt"]},
+        {algorithm: {name: "AES-KW", length: 128}, usages: ["wrapKey", "unwrapKey"]},
+        {algorithm: {name: "HMAC", length: 128, hash: "SHA-256"}, usages: ["sign", "verify"]}
+    ];
 
     // Generate all the keys needed, then iterate over all combinations
     // to test wrapping and unwrapping.
     promise_test(function() {
     return Promise.all([generateWrappingKeys(), generateKeysToWrap()])
     .then(function(results) {
-        var promises = [];
-        wrappers.forEach(function(wrapper) {
-            keys.forEach(function(key) {
-                promises.push(testWrapping(wrapper, key));
-            })
+        wrappingKeysParameters.forEach(function(wrapperParam) {
+            var wrapper = wrappers[wrapperParam.name];
+            keysToWrapParameters.forEach(function(toWrapParam) {
+                keys[toWrapParam.algorithm.name].forEach(function(key) {
+                    testWrapping(wrapper, key);
+                });
+            });
         });
-        return Promise.allSettled(promises);
+        return Promise.resolve("setup done");
+    }, function(err) {
+        return Promise.resolve("setup failed: " + err.name + ': "' + err.message + '"');
     });
```

Comment:
Should this really be a `Promise.resolve`? This will cause the test to pass, right?

```suggestion
    }).catch(function(err) {
        return Promise.reject("setup failed: " + err.name + ': "' + err.message + '"');
    });
```

Reply:
Umm, true. I think since the sub-tests are not going to be executed in any case I thought it was enough, but it's better indeed to finish it with 'reject'.